src/main/scala/com/microsoft/hyperspace/index/rules/IndexFilter.scala

```diff
@@ -29,16 +29,33 @@ import com.microsoft.hyperspace.util.{HyperspaceConf, ResolverUtils}
 
 trait IndexFilter extends ActiveSparkSession {
 
+  /**
+   * Append a given reason string to WHYNOT_REASON tag of the index if the condition is false and
+   * WHYNOT_ENABLED tag is set to the index.
+   *
+   * @param condition Flag for reason string
+   * @param reasonString Informational message in case condition is false.
+   * @param keyPair A pair of plan and index to tag
+   */
   protected def setReasonTag(condition: Boolean, reasonString: => String)(
       implicit keyPair: (LogicalPlan, IndexLogEntry)): Unit = {
```

Comment:
ditto on `implicit` here, and I don't think we have a caller using `implicit` yet.